pkg/cli/searcher.go

```diff
@@ -725,10 +725,35 @@ func (search referrerSearcher) search(config searchConfig) (bool, error) {
 	return true, printReferrersResult(config, referrersList, maxArtifactTypeLen)
 }
 
+const (
+	SortByAlphabeticASC = "alphabetic-asc"
+	SortByAlphabeticDSC = "alphabetic-dsc"
+	SortByLastUpdated   = "last-updated"
+	SortByRelevance     = "relevance"
+
+	SupportedCriterias = SortByAlphabeticASC + ", " + SortByAlphabeticDSC + ", " + SortByLastUpdated + ", " +
+		SortByRelevance
+)
+
+func userParamToGQLSortCriteria(param string) (string, error) {
+	switch param {
+	case SortByAlphabeticASC:
+		return "ALPHABETIC_ASC", nil
+	case SortByAlphabeticDSC:
+		return "ALPHABETIC_DSC", nil
+	case SortByLastUpdated:
+		return "UPDATE_TIME", nil
+	case SortByRelevance:
+		return "RELEVANCE", nil
+	default:
+		return "", zerr.ErrSortCriteriaNotSupported
+	}
+}
+
 type globalSearcherGQL struct{}
 
 func (search globalSearcherGQL) search(config searchConfig) (bool, error) {
-	if !canSearch(config.params, newSet("query")) {
+	if query := config.params["query"]; query == nil || *query == "" {
```

Comment:
I think we need a more generic way to handle mandatory vs optional parameters. There are other place where we could use this.

Reply:
I think we should specify the required parameters that determine a subcommand command. The can search command will now return true if the flag set contains at least the required parameters and it doesn't contains unsupported flags. The optional parameters will be given a default value if the flag is not present and if the flag is not part of the optional or required parameters then we'll return an error.

Reply:
The tricky part is in case a specific set of mandatory parameters match one search, and there's a different searcher which is support to run if only one of those is specified.

Your explanation sounds good, but we need to be careful.